src/http/client/client.cr

```diff
@@ -485,6 +511,10 @@ class HTTP::Client
     end
   end
 
+  private def self.allowed_scheme?(uri : URI)
+    uri.scheme =~ /https?/
```

Comment:
Probably `uri.scheme == "http" || uri.scheme == "https"` is way better and faster (no need to allocate a regex, to create a match data, etc.). Inside the standard library we try to be as efficient as possible. Also faster is just embedding the condition inside the method. Also, the same logic is done in `self.exec`, so we can maybe move it to a `private def self.ssl?(uri) : Bool` that checks that the scheme is non-nil, and it's only "http" or "https", and returns `true` if it's "https", `false` otherwise. What do you think?